inc/functions_upload.php

```diff
@@ -679,6 +679,100 @@ function upload_attachment($attachment, $update_attachment=false)
 	return $ret;
 }
 
+/**
+ * Process adding attachment(s) when the "Add Attachment" button is pressed.
+ *
+ * @param int $pid The ID of the post.
+ * @param array $forumpermission The permissions for the forum.
+ * @param string $attachwhere Search string "pid='$pid'" or "posthash='".$db->escape_string($mybb->get_input('posthash'))."'"
+ * @param string $action Where called from: "newthread", "newreply", or "editpost"
+ *
+ * @return array Array of errors if any, empty array otherwise
+ */
+function add_attachments($pid, $forumpermissions, $attachwhere, $action=false)
+{
+	global $db, $mybb, $editdraftpid, $lang;
+
+	$ret = array();
+
+	if($forumpermissions['canpostattachments'])
+	{
+		$attachments = array();
+		$fields = array ('name', 'type', 'tmp_name', 'error', 'size');
+
+		$total = isset($_FILES['attachments']['name']) ? count($_FILES['attachments']['name']) : 0;
+		$filenames = "";
+		$delim = "";
+		for($i=0; $i<$total; ++$i)
+		{
+			foreach($fields as $field)
+			{
+				$attach1[$field] = $_FILES['attachments'][$field][$key];
+				$attachments[$i][$field] = $_FILES['attachments'][$field][$i];
+			}
+
+			$FILE = $attachments[$i];
+			if(!empty($FILE['name']) && !empty($FILE['type']) && $FILE['size'] > 0)
+			{
+				$filenames .= $delim . "'" . $db->escape_string($FILE['name']) . "'";
+				$delim = ",";
+			}
+
+		}
+
+		$query = $db->simple_select("attachments", "filename, aid", "{$attachwhere} AND `filename` IN (".$filenames.")");
```

Comment:
Backticks (\`) around `filename` break Postgres with the following error:

```
0 - ERROR: operator does not exist: ` character varying LINE 1: ...E posthash='30d9dd80a5391cdb895b43fd4d844326' AND `filename`... ^ HINT: No operator matches the given name and argument type(s). You might need to add explicit type casts.
```

The name `filename` isn't a reserved word in mySQL, Postgres or SQLite as far as I can tell so I'm just going to remove these backpacks and continue testing.